Tasks/DeployAzureResourceGroup/task.json

```diff
@@ -203,12 +209,44 @@
         },
         {
             "name": "enableDeploymentPrerequisites",
-            "type": "boolean",
+            "type": "pickList",
             "label": "Enable deployment prerequisites",
-            "defaultValue": "false",
-            "required": false,
+            "defaultValue": "None",
             "groupName": "AdvancedDeploymentOptions",
-            "helpMarkDown": "Enabling this option configures Windows Remote Management (WinRM) listener over HTTPS protocol on port 5986, using a self-signed certificate. This configuration is required for performing deployment operation on Azure machines. If the target Virtual Machines are backed by a Load balancer, ensure Inbound NAT rules are configured for target port (5986)."
+            "options": {
+                "None": "None",
+                "ConfigureVMwithWinRM": "Configure virtual machines with WinRM",
+                "ConfigureVMWithMGAgent": "Configure virtual machines with machine group agent"
+            },
+            "helpMarkDown": "These options would be applicable only when the Resource group contains virtual machines. <br><br>Selecting WinRM option configures Windows Remote Management (WinRM) listener over HTTPS protocol on port 5986, using a self-signed certificate. This configuration is required for performing deployment operation on Azure machines. If the target Virtual Machines are backed by a Load balancer, ensure Inbound NAT rules are configured for target port (5986)."
+          
+        },
+        {
+            "name": "machineGroupName",
+            "type": "pickList",
+            "label": "Machine group",
+            "groupName": "MachineGroupOptions",
+            "helpMarkDown": "Select a machine group",
+            "properties": {
+                "DisableManageLink": "True"
+            }
+        },
+        {
+            "name": "copyAzureVMTags",
+            "type": "boolean",
+            "label": "Copy Azure VM tags to agents",
+            "groupName": "MachineGroupOptions",
+            "defaultValue": true,
+            "helpMarkDown": "Choose if the tags configured on the Azure VM need to be copied to the corresponding machine group agent. <br><br>​By default all Azure tags will be copied following the format “Key: Value”. Example: An Azure Tag “Role : Web” would be copied  as-is to the Agent machine. <br><br>For more information on how tag Azure resources refer to [link](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags​)"
+        },
+        {
+            "name": "vstsPATToken",
+            "type": "string",
+            "label": "Personal access token",
+            "groupName": "MachineGroupOptions",
+            "defaultValue": "",       
+            "required": true,
+            "helpMarkDown": "Agent registration with machine group requires access to your visual studio project.​"
```

Comment:
It is recommended that you store the token in a secure variable and provide the variable reference here.